auth/auth-server/src/telemetry/helpers.rs

```diff
@@ -148,9 +201,66 @@ pub fn record_external_match_metrics(
     }
 
     // Record response metrics
-    if let Err(e) = record_external_match_response_metrics(&match_resp, &labels) {
+    if let Err(e) = record_external_match_response_metrics(&match_resp.match_bundle, &labels) {
         warn!("Error recording response metrics: {e}");
     }
 
+    // Record settlement metrics
+    let did_settle = await_settlement(&match_resp.match_bundle, arbitrum_client).await?;
+    if let Err(e) =
+        record_external_match_settlement_metrics(&match_resp.match_bundle, did_settle, &labels)
+    {
+        warn!("Error recording settlement metrics: {e}");
+    }
+
     Ok(())
 }
+
+/// Await the result of the atomic match settlement to be submitted on-chain
+///
+/// Returns `true` if the settlement succeeded on-chain, `false` otherwise
+async fn await_settlement(
+    match_bundle: &AtomicMatchApiBundle,
+    arbitrum_client: &ArbitrumClient,
+) -> Result<bool, AuthServerError> {
+    let nullifier = extract_nullifier_from_match_bundle(match_bundle)?;
+    let res = arbitrum_client.await_nullifier_spent(nullifier, ATOMIC_SETTLEMENT_TIMEOUT).await;
+
+    let did_settle = res.is_ok();
+    if !did_settle {
+        warn!("atomic match settlement not observed on-chain");
```

Comment:
can we change this to `info`? this is the common case at the moment and it doesn't directly indicate that anything is wrong. been meaning to do that in the relayer